woocommerce-payments.php

```diff
@@ -333,7 +334,7 @@ function wcpay_get_jetpack_idc_custom_content(): array {
 	$urls = Automattic\Jetpack\Identity_Crisis::get_mismatched_urls();
 	if ( false !== $urls ) {
 		$current_url = untrailingslashit( $urls['current_url'] );
-		$wpcom_url   = untrailingslashit( $urls['wpcom_url'] );
+		$wpcom_url   = untrailingslashit( WC_Payments_Utils::reverse_url_if_trailing_slash( $urls['wpcom_url'] ) );
```

Comment:
```suggestion
        $idc_sync_error = Automattic\Jetpack\Identity_Crisis::check_identity_crisis();
        if ( is_array( $idc_sync_error ) && ! empty( $idc_sync_error['reversed_url'] ) ) {
			$urls['wpcom_url'] = strrev( $urls['wpcom_url'] );
		}
		$wpcom_url   = untrailingslashit( $urls['wpcom_url'] );
```

Reply:
@ismaeldcom WDYT?

Reply:
This solution looks better, but as I mentioned in Slack this morning, all we can do is patch a hacky solution. The real solution should be done on the Jetpack side.

Reply:
@vladolaru great solution, works like a charm! Done in https://github.com/Automattic/woocommerce-payments/pull/8033/commits/3a1d38fa33d94d4368cddf8622e2d6904a7d919e